### ISSUE TITLE
[Core] Support version skew between ray client & server

### ISSUE THREAD

**Issue by jhasm-ck** [ray-project/ray#37960]
### Description

Ray requires the client (ray python library installed using `pip install ray`) and server (ray cluster) versions to match exactly. This is good for consistency, but makes it impossible to upgrade ray version without a downtime. Essentially it requires the client+server to be upgraded atomically. This increases the blast radius of upgrades to all ray clusters, when the ray client is shared. This can potentially result in multiple attempts and multiple downtime periods.

An example of ecosystem that has solved this problem is kubernetes. See https://kubernetes.io/releases/version-skew-policy/. Since ray versions are upgraded frequently (for good reasons) it will be a great feature to be able to upgrade without too much trouble.

### Use case

1. Upgrade ray version of the long running ray clusters.
2. Upgrade ray version of the ray clusters hosting ray serve applications.
3. Upgrade ray version of multiple ray clusters accessed by the same client.

The lack of this feature is a blocker for running high availability services (e.g. models served through ray-serve), since we cannot upgrade the ray version without a downtime and we cannot accept scheduled downtimes for the required service availability.

**Comment by jhasm-ck** [comment]
@richardliaw

**Comment by rkooo567** [comment]
We are encouraging people to use job submission (https://docs.ray.io/en/master/cluster/running-applications/job-submission/index.html) due to some of innate limitations from ray client.

Besides, we'd like to understand why people use ray client better vs job submission. What are you using the ray client for, and is using ray job submission a viable path for you?

**Comment by richardliaw** [comment]
Hey @rkooo567 actually @jhasm-ck is using Job submission in this case. What he means is for the Job Submission Client to have a version skew tolerance with the Ray Cluster.

**Comment by jhasm-ck** [comment]
Yes correct, I mean the `ray job` command or a python client ([pip install ray](https://pypi.org/project/ray/)) used to interact with ray cluster and jobs. 
For my clarity, do you not call that ray client? I can make that clear in the future communications, if you use a different terminology.

**Comment by richardliaw** [comment]
Yeah we tend to just call it the Job Submission Client; Ray Client is another mechanism of submitting jobs to Ray (that we are deemphasizing)
